File: 백준/Silver/20044. Project Teams/Project Teams.py
import sys

n = int(sys.stdin.readline())
slist = list(map(int, sys.stdin.readline().split()))

slist.sort()

semi = 10000000
for i in range (n) :
    # print (slist[i], slist[-1-i])
    
    if slist[i] + slist[-1-i] < semi : 
        semi = slist[i] + slist[-1-i]
        
    else :
        pass
    
print (semi)

File: 백준/Silver/20044. Project Teams/Project Teams.cc
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<int> slist(2 * n); 

    for (int i = 0; i < 2 * n; i++) {
        cin >> slist[i];
    }

    sort(slist.begin(), slist.end()); 

    int semi = INT_MAX;
    for (int i = 0; i < n; i++) {
        semi = min(semi, slist[i] + slist[2 * n - 1 - i]);
    }

    cout << semi << '\n'; 

    return 0;
}